source/game.mission.bmx

```diff
@@ -550,4 +558,188 @@ Type TCombinedMission extends TMission
+		return create(ALL_TYPES, 0, days)
+	End Function
+
+	Function createInARow:TSammyMission(count:Int, days:Int=-1)
+		return create(IN_A_ROW, count, days)
+	End Function
+
+	Function create:TSammyMission(mType:String, targetValue:Int=-1, days:Int=-1)
+		Local result:TSammyMission=new TSammyMission
+		result.missionType = mType
+		result.targetValue = targetValue
+		result.daysForAchieving = days
+		return result
+	End Function
+
+	Method getCheckListeners:TEventListenerBase[]()
+		Return [ EventManager.registerListenerMethod(GameEventKeys.Award_OnFinish, Self, "OnAward") ]
+	EndMethod
+
+	Method getTitle:String()
+		Return GetLocale("MISSION_TITLE_"+getCategory())
+	End Method
+
+	Method getCategory:String()
+		return "SAMMY"
+	End Method
+
+	Method getIdSuffix:String() override
+		Local mapName:String = GetStationMapCollection().GetMapName()
+		Local suffix:String = mapName + "_" + missionType
+		If targetValue > 0 Then suffix:+("_"+targetValue)
+		Return suffix
+	End Method
+
+	Method getDescription:String()
+		Local desc:String = GetLocale("MISSION_GOAL_SAMMY_"+missionType)
+		If missionType = IN_A_ROW Then desc = desc.Replace("%COUNT%", targetValue)
+
+		If daysForAchieving > 0
+			If targetValue < 0 
+				desc = GetLocale("MISSION_TIME_AFTER").Replace("%GOAL%", desc).replace("%DAYS%",daysForAchieving)
+			Else
+				desc = GetLocale("MISSION_TIME_WITHIN").Replace("%GOAL%", desc).replace("%DAYS%",daysForAchieving)
+			EndIf
+		EndIf
+		Return desc
+	EndMethod
+
+
+	Method OnAward:Int(triggerEvent:TEventBase)
+		currentAward:TAward = TAward(triggerEvent.GetSender())
+		If Not currentAward Then throw "TSammyMission.Award: no award in event"
+		checkMissionResult(False)
+	End Method
+
+	Method checkMissionResult(forceFinish:Int=False)
+		Local awards:TAwardCollection = GetAwardCollection()
+		If Not awards Then Throw "TSammyMission.checkMissionResult: no awards collection present"
+		If currentAward
+			If awards.currentAward And currentAward <> awards.currentAward Then print "TSammyMission.checkMissionResult: award from event is not the one currently awarded"
+		EndIf
+		Local list:TList = awards.lastAwards
+		'create copy in reversed order
+		If list
+			list = list.Reversed()
+			If currentAward And list.contains(currentAward) Then throw "TSammyMission.checkMissionResult: last awards not expected to contain current Sammy, yet"
+		Else
+			list = new TList
+		EndIf
+		If currentAward Then list.addFirst(currentAward)
+		'at this point list is the full list of awards with the latest as first element
+
+		'check winning condition
+		Local currentPlayer:Int = -1
+		Local winningPlayer:Int = 0
+		If missionType <> MAXIMIZE And currentAward Then currentPlayer = currentAward.winningPlayerID
+		Local key:TEventKey = GameEventKeys.Mission_Achieved
+		Local fireEvent:Int = False
+		Local text:String = ""
+		Select missionType
+			Case IN_A_ROW
+				Local count:Int = 0
+				For Local aw:TAward = EachIn list
+					If aw
```

Comment:
An sich braucht es kein "if aw", da das `For Local aw:TAward ...` schon nur Objekte in der Liste rausrueckt, die vom Typ "TAward" (oder darauf basierend) sind.

Reply:
OK. D.h. selbst wenn andere Objekte oder null-Werte drinwären, kann ich mir bei EachIn sicher sein, dass der Typ immer passt?